File: rest/src/main/java/io/sparkled/viewmodel/stage/search/StageSearchViewModel.java
package io.sparkled.viewmodel.stage.search;

import io.sparkled.viewmodel.ViewModel;

import java.util.Objects;

public class StageSearchViewModel implements ViewModel {
    private Integer id;
    private String name;

    public Integer getId() {
        return id;
    }

    public StageSearchViewModel setId(Integer id) {
        this.id = id;
        return this;
    }

    public String getName() {
        return name;
    }

    public StageSearchViewModel setName(String name) {
        this.name = name;
        return this;
    }


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        StageSearchViewModel that = (StageSearchViewModel) o;
        return Objects.equals(id, that.id) &&
                Objects.equals(name, that.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, name);
    }

    @Override
    public String toString() {
        return "StageSearchViewModel{" +
                "id=" + id +
                ", name='" + name + '\'' +
                '}';
    }
}

File: rest/src/main/kotlin/io/sparkled/viewmodel/stage/search/StageSearchViewModel.kt
package io.sparkled.viewmodel.stage.search

import io.sparkled.viewmodel.ViewModel

class StageSearchViewModel : ViewModel {
    private var id: Int? = null
    private var name: String? = null

    fun getId(): Int? {
        return id
    }

    fun setId(id: Int): StageSearchViewModel {
        this.id = id
        return this
    }

    fun getName(): String? {
        return name
    }

    fun setName(name: String): StageSearchViewModel {
        this.name = name
        return this
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as StageSearchViewModel

        if (id != other.id) return false
        if (name != other.name) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id ?: 0
        result = 31 * result + (name?.hashCode() ?: 0)
        return result
    }

    override fun toString(): String {
        return "StageSearchViewModel(id=$id, name=$name)"
    }
}
